src/common/tools/gen_listings_and_webworkers.ts

```diff
@@ -78,12 +78,62 @@ export const listing = ${JSON.stringify(listing, undefined, 2)};
         `\
 // AUTO-GENERATED - DO NOT EDIT. See ${myself}.
 
-// oldG is a TestGroup<Fixture> object (defined in common/internal/test_group.ts).
-import { g as oldG } from '${relPathToSuiteRoot}/${entry.file.join('/')}.spec.js';
+// g is a TestGroup<Fixture> object (defined in common/internal/test_group.ts).
+import { g } from '${relPathToSuiteRoot}/${entry.file.join('/')}.spec.js';
+
+import { globalTestConfig } from '/out/common/framework/test_config.js';
+import { Logger } from '/out/common/internal/logging/logger.js';
+import { setDefaultRequestAdapterOptions } from '/out/common/util/navigator_gpu.js';
+import { parseQuery } from '/out/common/internal/query/parseQuery.js';
+import { comparePublicParamsPaths, Ordering } from '/out/common/internal/query/compare.js';
+import { assert } from '/out/common/util/util.js';
+
+async function reportTestResults(ev) {
+  const query = ev.data.query;
+  const expectations = ev.data.expectations;
+  const ctsOptions = ev.data.ctsOptions;
+
+  const { debug, unrollConstEvalLoops, powerPreference, compatibility } = ctsOptions;
+  globalTestConfig.unrollConstEvalLoops = unrollConstEvalLoops;
+  globalTestConfig.compatibility = compatibility;
+
+  Logger.globalDebugMode = debug;
+  const log = new Logger();
+
+  if (powerPreference || compatibility) {
+    setDefaultRequestAdapterOptions({
+      ...(powerPreference && { powerPreference }),
+      // MAINTENANCE_TODO: Change this to whatever the option ends up being
+      ...(compatibility && { compatibilityMode: true }),
+    });
+  }
```

Comment:
This looks copy-pasted, I think? Could it be factored out into a function to avoid duplicated code?

Reply:
The `reportTestResults()` function used by dedicated workers and shared workers in `src/common/runtime/helper/test_worker-worker.ts` is similar in the first part, but not in how handle discovering the test case and running it.
I didn't find a way to avoid duplication code.

Reply:
I mean just the selection of code I put this comment on - from `const ... = ctsOptions` to `setDefaultRequestAdapterOptions`.
This part has to be updated every tim ewe add new global test/adapter options so it would be nice to have it in one place.